### PR TITLE
Bump versions of wasm-tools crates

### PR DESCRIPTION
**Pull request by alexcrichton**
Primarily also release the `wit-parser` and `wit-component` crates to crates.io.

### CODE REVIEW

**Review by peterhuene — APPROVED** [review]
Should we include the `decode_world` change in this release?

**Comment by peterhuene** [comment]
Also, I parked `wit-component` on crates.io, so I'll need to transfer that over to the BCA.

**Comment by alexcrichton** [comment]
Indeed! I'll wait for that to merge first before landing this.

**Pull request merged**
